### PR TITLE
Pytest driven benchmarks

### PR DESCRIPTION
**Pull request by HDembinski**
I added some pytest-based benchmarks of the various algorithms we use (bootstrap algorithms, confidence interval calculation). The results are very interesting. This complementary to your approach of benchmarking the performance over time. It gives us data on which algorithms to recommend and pick as the default.

For example, I found that the computation of studentized confidence intervals is as fast as "ordinary", which I think is an argument to make that the default. What do you think?

### CODE REVIEW

**Comment by dsaxton** [comment]
I wasn't familiar with pytest benchmarks, is there a way to use this to take a "benchmark diff" of sorts to detect regressions? Actually on the surface this seems much lighter weight and cleaner than asv and if this works out I wouldn't mind getting rid of the latter in favor of this approach.

**Comment by dsaxton** [comment]
Can you resolve the merge conflicts? Also the current asv benchmark setup is out of sync with the recent changes so we should just decide here if we want to remove them.

**Comment by HDembinski** [comment]
I think pytest-benchmark is sufficient for our purposes. We don't have a project like PyPy with many parts that work together, where minor changes can causes a regression elsewhere. When we make something faster in this simple library, it will have an isolated effect. I like about pytest-benchmark that it allows me to generate reports easily that compare the different methods that we have implemented. It also makes it easy to test a new method, by implementing it directly in the test_X.py script.

I made it so that the benchmarks are not automatically run together with the other tests, when you do `pytest` in the main directory, because they take some time. Unit tests should execute fast. On CI, however, the benchmarks should be run routinely, to make sure that we do not break them accidentally.

pytest-benchmark supports storing and comparing the benchmark reports.
https://pytest-benchmark.readthedocs.io/en/latest/

**Comment by HDembinski** [comment]
I think it should be possible to generate a report and publish this as an artifact.

**Comment by dsaxton** [comment]
> I think pytest-benchmark is sufficient for our purposes.

I think we can just remove asv benchmarks in this PR then.

> I made it so that the benchmarks are not automatically run together with the other tests, when you do `pytest` in the main directory, because they take some time.

How long do they take to run? Unless they're hours it may not be too bad to add them to CI. They could also be limited to commits to master instead of all PR commits.

**Comment by HDembinski** [comment]
Running the benchmarks takes 30 sek on my machine, so not too long for CI, just too long for unit testing during active development. That's why I set it up so that these tests are not included when you run `pytest` in project directory. To run them, you need to explicitly do `pytest benchmarks` in the project path.

File: requirements-dev.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+numpy>=1.17
+scipy>=1.5
+numba>=0.49.1
```
Comment (dsaxton) [inline requirements-dev.txt:3]: Remove numba?

Comment (HDembinski) [reply requirements-dev.txt:3]: Right, that's wrong now.

**Pull request merged**